cardano-api/test/Test/Cardano/Api/Eras.hs

```diff
@@ -0,0 +1,50 @@
+module Test.Cardano.Api.Eras
+  ( tests
+  ) where
+
+import           Cardano.Api
+import           Cardano.Api.Orphans ()
+import           Data.Aeson (ToJSON (..), decode, encode)
+import           Hedgehog (Property, forAll, property, (===))
+import qualified Hedgehog.Gen as Gen
+import           Test.Tasty (TestTree, testGroup)
+import           Test.Tasty.Hedgehog (testPropertyNamed)
+
+--------------------------------------------------------------------------------
+-- Bounded instances
+
+prop_maxBound_CardanoMatchesShelley :: Property
+prop_maxBound_CardanoMatchesShelley = property $ do
+  AnyCardanoEra era <- forAll $ Gen.element [maxBound]
+  AnyShelleyBasedEra sbe <- forAll $ Gen.element [maxBound]
+
+  fromEnum (AnyCardanoEra era) === fromEnum (AnyCardanoEra (shelleyBasedToCardanoEra sbe))
+
+--------------------------------------------------------------------------------
+-- Aeson instances
+
+prop_roundtrip_JSON_Shelley :: Property
+prop_roundtrip_JSON_Shelley = property $ do
+  AnyShelleyBasedEra sbe <- forAll $ Gen.element [minBound..maxBound]
```

Comment:
why not use `tripping` from the hedgehog library?

Reply:
Fixed.  Thanks!